### ISSUE TITLE
MIT License

### ISSUE THREAD

**Issue by rafaltrus** [rafalio/dancing-links-java#1]
Can I use your code in my commercial product? I think so since it's under MIT License.. But I want to confirm. Thanks! Great work!

**Comment by rafalio** [comment]
Yes, as it's MIT license. You just need to make sure it's credited somewhere in the code with a copy of the license.
Let me know how you used it, I'm interested.

**Comment by rafaltrus** [comment]
I am looking to implement this as a back-bone of my project:

http://www.math.utah.edu/~yzhang/teaching/1030/Sudoku.pdf

Very interesting paper.

**Comment by rafalio** [comment]
Is this your paper? It's quite hard to follow.

**Comment by rafaltrus** [comment]
Nope. I found it during my research. I am trying to apply Algorithm X and Dancing Links to other places, but I first need to understand/implement them first.

https://www.kth.se/social/files/54bda0d3f276541354ec0425/HLaestanderMHarrysson_dkand14.pdf

This paper, from Royal Institute of Technology, University in Stockholm, Sweden, is also on-topic.

**Comment by rafalio** [comment]
Yeah this is way better, forget the other paper, don't even read it. 
This one is good but spends 30 pages on something not very complicated though and takes the weird approach to measure runtime in units of time instead of algorithm steps (and also with Python, a garbage collected language).
Just read the Knuth paper (or as much as you need of it, it then goes on about pentominoes), and you should easily figure out the transformation to solve a sudoku with DLX. It's somewhere in my code also if I recall right.

**Comment by rafaltrus** [comment]
Would you happen to have a direct link to that paper? I can't find it online..

**Comment by rafaltrus** [comment]
"a garbage collected language".. ha!

**Comment by rafalio** [comment]
https://jnucsot.googlecode.com/files/dancing-color.pdf

**Comment by rafaltrus** [comment]
Thanks, mate!
